scripts/update_license_headers.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+import re
+import sys
+
+def update_license_header(file_path):
+    with open(file_path, 'r') as file:
+        content = file.read()
+
+    apache_header_pattern = re.compile(
+        r'(/\*|#|//)\s*Licensed under the Apache License, Version 2\.0.*?'
+        r'limitations under the License\.',
+        re.DOTALL
+    )
+    spdx_header = "// SPDX-License-Identifier: Apache-2.0\n"
+
+    if content.startswith(spdx_header):
```

Comment:
the first line should always be Copyright, so this should never match

Reply:
also, why does it matter - if you don't find Apache header you skip the file, irrespective of it having SPDX

Reply:
I have updated the code and tested it locally again.
 - It correctly preserves all copyright lines.
 - successfully replaces the Apache license text with the SPDX identifier.
 - empty comment lines between copyright notices and license text are removed.